plugin/keyring_vault/CMakeLists.txt

```diff
@@ -0,0 +1,56 @@
+CHECK_CXX_COMPILER_FLAG(-std=c++11 HAVE_STDCXX11)
+
+IF (NOT WITH_CURL_INCLUDE)
```

Comment:
I have changed this implementation - I have created a new macro for finding CURL. Now keyring_vault can be build with system CURL. I did not want user to be able to build easily with system CURL for two reasons:
- by default - most of the distros ship CURL with rtmp library build with CURL. When we are linking CURL to keyring_vault there can be undefined symbols left so no problem - we link only with Curl lib and forget about rtmp. However, when I link unitest then apart from CURL lib I need to specify also rtmp lib so the unit tests would build. Thus if one wants to build unitest he/she cannot build CURL with rtmp support.
- second issue is when CURL is linked with system openssl while Server was built with custom ssl, then we get a segfault

In short it is hard to make CURL to behave nicely with MySQL. This is the reason I believe, Oracle builds CURL statically and link it statically with the Server, which is what I think we should also be doing.